File: packages/dotnet/src/SlyBrowser/SlyBrowserOptions.cs
namespace SlyBrowser;

/// <summary>User-facing options shared with the Node.js, Python and Java SDKs.</summary>
public sealed class SlyBrowserOptions
{
    public IReadOnlyDictionary<string, object?> Profile { get; init; } =
        new Dictionary<string, object?>(StringComparer.Ordinal);
    public LaunchOptions Launch { get; init; } = new();
    public HumanizeOptions Humanize { get; init; } = new();
}

public sealed class LaunchOptions
{
    public bool Headless { get; init; } = true;
    public string ProfileMode { get; init; } = "ephemeral";
    public string? ProfileDirectory { get; init; }
    public bool UpdateKernel { get; init; } = true;
}

public sealed class HumanizeOptions
{
    public bool Enabled { get; init; }
    public string Preset { get; init; } = "default";
    public int? Seed { get; init; }
    public IReadOnlyDictionary<string, double>? Config { get; init; }
}

File: packages/java/src/main/java/com/slybrowser/SlyBrowserOptions.java
package com.slybrowser;

import java.nio.file.Path;
import java.util.Collections;
import java.util.Map;

/** User-facing options shared with the Node.js, Python and .NET SDKs. */
public final class SlyBrowserOptions {
  public Map<String, Object> profile = Collections.emptyMap();
  public LaunchOptions launch = new LaunchOptions();
  public HumanizeOptions humanize = new HumanizeOptions();

  public static final class LaunchOptions {
    public boolean headless = true;
    public String profileMode = "ephemeral";
    public Path profileDirectory;
    public boolean updateKernel = true;
  }

  public static final class HumanizeOptions {
    public boolean enabled;
    public String preset = "default";
    public Integer seed;
    public Map<String, Number> config;
  }
}
